buildkite/scripts/test-nix.sh

```diff
@@ -67,7 +67,7 @@ git reset --hard $BUILDKITE_COMMIT
 nix "${NIX_OPTS[@]}" build "$PWD?submodules=1#devnet" --no-link
 
 # Test developer terminal with lsp server
-nix "${NIX_OPTS[@]}" develop "$PWD?submodules=1#with-lsp" --command bash -c "echo tested"
+nix "${NIX_OPTS[@]}" develop "$PWD?submodules=1#with-lsp" --command bash -c "dune build src"
```

Comment:
I'd approach this test differently:

* `nix "${NIX_OPTS[@]}" develop "$PWD?submodules=1#with-lsp" --command bash -c "echo tested"`
* `nix "${NIX_OPTS[@]}" develop "$PWD?submodules=1#with-lsp" --command bash -c "dune build src" &`
* `nix "${NIX_OPTS[@]}" build "$PWD?submodules=1#devnet" --no-link`
* `wait`, to ensure the `dune build src` finishes

Otherwise it would result in two sequential builds of Mina going one by one, which may take a long of time.

Reply:
We could also just remove this change from the PR and attempt it in a different PR.

Reply:
I made your suggested changes and it seems to be working fine now